### PR TITLE
[Fix] Fix two significant bugs in SOT and provide new SOT pretrained models

### PR DESCRIPTION
**Pull request by JingweiZhang12**
This PR mainly contains the following:
1. **Fix bug: ** The gradient of some SiamRPN++ parameters initially fixed are out of sync in DDP after they are required updated halfway through training.
2. **Fix bug: ** Considering the performance fluctuation in SOT,  we set the fixed seed, deterministic=True and cudnn_benchmark=False in SOT configs.
3. Fix the typos about fp16 path.
4. Provide new pretrained models for SiameseRPN++

### CODE REVIEW

File: configs/sot/siamese_rpn/siamese_rpn_r50_1x_lasot.py [hunk 1]
```diff
@@ -201,3 +203,4 @@
 load_from = None
 resume_from = None
 workflow = [('train', 1)]
+find_unused_parameters = True
```
Comment (GT9505) [inline configs/sot/siamese_rpn/siamese_rpn_r50_1x_lasot.py:206]: put `find` in the front of the file

File: mmtrack/models/backbones/sot_resnet.py [hunk 2]
```diff
@@ -164,6 +164,22 @@ class SOTResNet(ResNet):
...
         super(SOTResNet, self).__init__(depth, *args, **kwargs)
+        self._unfreeze_stages()
+
+    def _unfreeze_stages(self):
+        if self.frozen_stages >= 0:
+            if self.deep_stem:
+                for param in self.stem.parameters():
+                    param.requires_grad = True
+            else:
+                for m in [self.conv1, self.norm1]:
```
Comment (GT9505) [inline mmtrack/models/backbones/sot_resnet.py:175]: set bn to `train()` mode

File: configs/sot/siamese_rpn/README.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ We observe around 1.0 points fluctuations in Success and 1.5 points fluctuations
...
 | SiamRPN++ |    R-50    |  -  |   20e    | 7.54        | 50.0              | 49.9 | 57.9 | [config](siamese_rpn_r50_1x_lasot.py) | [model](https://download.openmmlab.com/mmtracking/sot/siamese_rpn/siamese_rpn_r50_1x_lasot/siamese_rpn_r50_1x_lasot_20201218_051019-3c522eff.pth) &#124; [log](https://download.openmmlab.com/mmtracking/sot/siamese_rpn/siamese_rpn_r50_1x_lasot/siamese_rpn_r50_1x_lasot_20201218_051019.log.json) |
-| SiamRPN++ <br> (FP16) |    R-50    |  -  |   20e    | -        | -              | 49.1 | 57.0 | [config](siamese_rpn_r50_fp16_1x_lasot.py) | [model](https://download.openmmlab.com/mmtracking/siamese_rpn_r50_fp16_1x_lasot_20210731_110245-6733c67e.pth) &#124; [log](https://download.openmmlab.com/mmtracking/siamese_rpn_r50_fp16_1x_lasot_20210731_110245.log.json) |
+| SiamRPN++ <br> (FP16) |    R-50    |  -  |   20e    | -        | -              | 49.1 | 57.0 | [config](siamese_rpn_r50_fp16_1x_lasot.py) | [model](https://download.openmmlab.com/mmtracking/fp16/siamese_rpn_r50_fp16_1x_lasot_20210731_110245-6733c67e.pth) &#124; [log](https://download.openmmlab.com/mmtracking/fp16/siamese_rpn_r50_fp16_1x_lasot_20210731_110245.log.json) |
```
Comment (GT9505) [inline configs/sot/siamese_rpn/README.md:52]: remove `round` operation in sot, and check whether the performance of training and testing is consistent

**Review by GT9505 — APPROVED** [review]

File: mmtrack/models/backbones/sot_resnet.py [hunk 4]
```diff
@@ -164,6 +164,27 @@ class SOTResNet(ResNet):
     def __init__(self, depth, *args, **kwargs):
         assert depth == 50, 'Only support r50 backbone for sot.'
         super(SOTResNet, self).__init__(depth, *args, **kwargs)
+        # unfreeze the backbone parameters so that DDP can build all parameters
+        # into buckets.
+        self._unfreeze_stages()
```
Comment (GT9505) [inline mmtrack/models/backbones/sot_resnet.py:169]: Do we need a bool variable (e.g. `unfreeze_backbone`) to control whether do `_unfreeze_stages`?

File: configs/sot/siamese_rpn/README.md [hunk 5]
```diff
@@ -44,12 +44,12 @@ VOT2018, UAV123, LaSOT, and TrackingNet.
...
-| SiamRPN++ <br> (FP16) |    R-50    |  -  |   20e    | -        | -              | 49.1 | 57.0 | [config](siamese_rpn_r50_fp16_1x_lasot.py) | [model](https://download.openmmlab.com/mmtracking/siamese_rpn_r50_fp16_1x_lasot_20210731_110245-6733c67e.pth) &#124; [log](https://download.openmmlab.com/mmtracking/siamese_rpn_r50_fp16_1x_lasot_20210731_110245.log.json) |
+| SiamRPN++ |    R-50    |  -  |   20e    | 7.54        | 50.0              | 50.1 | 59.1 | [config](siamese_rpn_r50_1x_lasot.py) | [model](https://download.openmmlab.com/mmtracking/sot/siamese_rpn/siamese_rpn_r50_1x_lasot/siamese_rpn_r50_1x_lasot_20211203_151612-da4b3c66.pth) &#124; [log](https://download.openmmlab.com/mmtracking/sot/siamese_rpn/siamese_rpn_r50_1x_lasot/siamese_rpn_r50_1x_lasot_20211203_151612.log.json) |
```
Comment (GT9505) [inline configs/sot/siamese_rpn/README.md:51]: Provide the results of `precision` metric

File: configs/sot/siamese_rpn/README.md [hunk 6]
```diff
@@ -62,28 +62,40 @@ The checkpoints from 10-th to 20-th epoch will be evaluated during training. You
...
+
+### VOT2018
+
+The checkpoints from 10-th to 20-th epoch will be evaluated during training. You can find the best checkpoint from the log file.
+
+If you want to get better results, you can use the best checkpoint to search the hyperparameters on VOT2018 following [here](https://github.com/open-mmlab/mmtracking/blob/master/docs/useful_tools_scripts.md#siameserpn-test-time-parameter-search).
+
+**Note:** We train the SiameseRPN++ in the official [pysot](https://github.com/STVIR/pysot) codebase and can not reproduce the same results reported in the paper. We only get 0.364 EAO by following the training and hyperparameters searching instructions of pysot, which are lower than those of the paper by 0.05 EAO. In our codebase, the EAO are lower 0.066 than those of the paper. Notably, the results below are achieved without hyperparameters search.
```
[the remark this answers is not in the document]
Comment (JingweiZhang12) [reply configs/sot/siamese_rpn/README.md:97]: yes

[the remark this answers is not in the document]
Comment (JingweiZhang12) [reply configs/sot/siamese_rpn/README.md:97]: Yes

**Review by GT9505 — APPROVED** [review]
